exercises/complex-numbers/description.md

```diff
@@ -1,29 +1,97 @@
 # Description
 
-A complex number is a number in the form `a + b * i` where `a` and `b` are real and `i` satisfies `i^2 = -1`.
+A **complex number** is expressed in the form `z = a + b * i`, where:
 
-`a` is called the real part and `b` is called the imaginary part of `z`.
-The conjugate of the number `a + b * i` is the number `a - b * i`.
-The absolute value of a complex number `z = a + b * i` is a real number `|z| = sqrt(a^2 + b^2)`. The square of the absolute value `|z|^2` is the result of multiplication of `z` by its complex conjugate.
+- `a` is the **real part** (a real number),
 
-The sum/difference of two complex numbers involves adding/subtracting their real and imaginary parts separately:
-`(a + i * b) + (c + i * d) = (a + c) + (b + d) * i`,
-`(a + i * b) - (c + i * d) = (a - c) + (b - d) * i`.
+- `b` is the **imaginary part** (also a real number), and
 
-Multiplication result is by definition
-`(a + i * b) * (c + i * d) = (a * c - b * d) + (b * c + a * d) * i`.
+- `i` is the **imaginary unit** satisfying `i^2 = -1`.
 
-The reciprocal of a non-zero complex number is
-`1 / (a + i * b) = a/(a^2 + b^2) - b/(a^2 + b^2) * i`.
+## Key Properties
```

Comment:
What makes the absolute value a property and not an operation?

Reply:
I believe the concept of "key property" isn't very well defined and can be interpreted in different ways. Technically, neither the conjugate nor the absolute value is regarded as a property of complex numbers. Properties refer to characteristics of the numbers, like the associative rule `(a + b) + c = a + (b + c)` for real numbers.

Personally, I'm not too picky about this, given that we're not really teaching math here, and what we already have should be sufficient for this exercise. However, if we want to be accurate, both the conjugate and the absolute value (or square) are simply operations.

Reply:
It sounds like you're defining "property" as a function which operates on one complex number and defining "operation" as a function which operates on two complex numbers. In that case, wouldn't "exponent" be a property? Or would it make more sense to not apply this distinction and just call them all "operations"?

Reply:
The key point I want to highlight is that operations like conjugate, absolute value, exponentiation, and reciprocal are unary operators, while addition, subtraction, multiplication, and division are binary operators for complex numbers. This raises an important question: should unary and binary operators be categorized separately, or should they all be grouped together under a unified "Operations" section?  

My reasoning for placing **absolute value** and **conjugate** in the *Key Properties* section, separate from other operations, is that these two play a foundational role in complex number arithmetic. For example:  
- **Absolute value** is central to defining magnitude, normalization, and division.  
- **Conjugate** is essential for computing reciprocals and simplifying division.  

I acknowledge, however, that both involve computation and are technically unary operations, not inherent "properties" in the strict mathematical sense. As @tasxatzial rightly pointed out, properties typically refer to inherent characteristics like associativity or commutativity.  

That said, my intention in categorizing them as "key properties" was to emphasize their foundational role in complex number operations. If unifying all unary and binary operators under "Operations" better suits the context of this repository, I’m happy to align with that approach.  

Looking forward to your thoughts on whether separating these foundational elements adds clarity or if a unified categorization would be more consistent.

Reply:
Since the exercise requires implementing 'operations', which include both the conjugate and the absolute value, it might be better to move them to the operations section for consistency.